### PR TITLE
add proposal of socks5proxy

### PR DESCRIPTION
**Pull request by lynxcat**
Add proposal document of sock5proxy

### CODE REVIEW

File: docs/proposal/edgemesh-ssh-proposal.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+The ssh client initiates a request through the socks5 proxy and proxy the traffic to the socks5 proxy.
+
+**step1:**  
+In the socks5 proxy, get the target host and target port. Convert the targat host to the target IP through k8s api. The edgemesh-agent on the node uses hostnetwork, so just take its pod id directly.
+
+**step2:**  
+Encapsulate the converted target ip and target port and transmit them to the tunnel module
+
+**step3:**  
+The subsequent tunnel module will use the original p2p channel to send the request to the target machine and request the corresponding service, and then return the traffic to socks5 to complete the traffic proxy
```
Comment (khalid-jobs) [inline docs/proposal/edgemesh-ssh-proposal.md:24]:
```suggestion
The subsequent tunnel module will use the original p2p channel to send the request to the target machine and request the corresponding service (SSH server), and then return the traffic to socks5 to complete the traffic proxy
```

Comment (lynxcat) [reply docs/proposal/edgemesh-ssh-proposal.md]: Thanks. I've revised it.

**Pull request merged**